File: 946.cpp
#include <stack>
#include <vector>

using namespace std;

class Solution {
public:
  bool validateStackSequences(vector<int> &pushed, vector<int> &popped) {
    int i = 0;
    stack<int> s;
    for (int elt : pushed) {
      s.push(elt);
      while (s.size() != 0 && i < popped.size() && s.top() == popped[i]) {
        s.pop();
        ++i;
      }
    }
    return s.size() == 0;
  }
};

File: 946.py
class Solution:
    def validateStackSequences(self, pushed: list[int], popped: list[int]) -> bool:
        st = []
        i = 0
        for elt in pushed:
            st.append(elt)
            while i < len(popped) and st and st[-1] == popped[i]:
                st.pop()
                i += 1
        return len(st) == 0


Solution().validateStackSequences([1, 0], [1, 0])
